File: code/0496 Next Greater Element I.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        def helper(num):
            for tmp in nums2[nums2.index(num):]:
                if tmp > num:
                    return tmp
            return -1
        return map(helper, nums1)

File: code/0496 Next Greater Element I.cpp
/**
 * Return an array of size *returnSize.
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* nextGreaterElement(int* nums1, int nums1Size, int* nums2, int nums2Size, int* returnSize) {
    int* result =(int*)malloc(nums1Size*sizeof(int));
    *returnSize = nums1Size;
    for (int i = 0; i < nums1Size ; i ++)
        for (int j = 0; j < nums2Size; j ++) {
            if (*(nums2 + j) == *(nums1 + i)) {
                int k = j + 1;
                for (; k < nums2Size; k++) {
                    if (*(nums2 + k) > *(nums2 + j)) {
                        *(result + i) = *(nums2 + k);
                        break;
                    }
                }
                if (k == nums2Size) {
                    *(result + i) = -1;
                }
                break;
            }
        }
    return result;
}
